### ISSUE TITLE
[BUG GeneralMixtureModel do not freeze weight]

### ISSUE THREAD

**Issue by jeammimi** [jmschrei/pomegranate#914]
**Describe the bug**
Hello,
I would like to use a GeneralMixtureModel of two GeneralMixtureModels (of gaussian distribution), g1 and g2.
The first one, g1 has been fit on a reference, and I would like to free all its parameters.
I managed to freeze the parameters of the gaussians inside g1, however I cannot freeze the weights of the mixture:
setting g1.frozen = True, did nothing.
To do this I hesitate between  i) creating a custom distribution and ii) modifying the code.
adding this in the fit function of the generalMixtureModels:
```
if self.frozen:
   return self
```
could this  work ?
Is there another way to do it?
Best

**Comment by organic-chemistry** [comment]
I found a workaround:
once g1 has been fit I freeze it defining this new class:
```python
class FrozenGeneralMixtureModel(GeneralMixtureModel):
    def __init__(self, gm):
        super( GeneralMixtureModel, self ).__init__(distributions=gm.distributions,
                                                   weights=np.exp(gm.weights))
    def from_summaries(self,*args,**kwargs):
        return self

frozen_g1 = FrozenGeneralMixtureModel(g1)
```
I think that it is a bug as GeneralMixtureModel has a frozen attribute and when you set it to True nothing append.

**Comment by jmschrei** [comment]
This should be fixed in 0.14.6. Please re-open if the problem persists.
